apps/frontend/src/components/global/checklist/ChecklistRulesTable.vue

```diff
@@ -0,0 +1,231 @@
+          </strong>
+        </div>
+        <div style="width: fit-content">
+          <v-switch
+            input-value="true"
+            style="margin-top: 0; padding-top: 0"
+            dense
+            inset
+            label="Short ID"
+            color="teal"
+            hide-details
+            @change="$emit('toggle-short-id')"
+          />
+        </div>
+      </div>
+      <v-select
+        v-model="selectedHeaders"
+        :items="headersList"
+        label="Select Columns"
+        class="mt-4 pt-0"
+        dark
+        item-color="white"
+        multiple
+        outlined
+        return-object
+        height="8.5vh"
+      >
+        <template #selection="{item, index}">
+          <v-chip v-if="index < 3" small>
+            <span>{{ item.text }}</span>
+          </v-chip>
+          <span v-if="index === 3" class="grey--text caption ml-2">
+            (+{{ selectedHeaders.length - 3 }} others)
+          </span>
+        </template>
+      </v-select>
+    </v-card-title>
+    <v-card-text>
+      <!-- The mobile-breakpoint attribute fixes issues with the mobile resizing of a v-data-table-->
+      <v-data-table
+        :single-select="true"
+        disable-pagination
+        dense
+        mobile-breakpoint="0"
+        :items="rules"
+        :item-class="checkSelected"
+        :headers="headers"
+        hide-default-footer
+        class="overflow-auto"
+        height="68vh"
+        @current-items="getFiltered"
+        @click:row="showRule"
+      >
+        <template #[`item.status`]="{item}">
+          <v-chip :color="statusColor(item.status)" small>
+            <strong>{{ shortStatus(item.status) }}</strong>
+          </v-chip>
+        </template>
+        <template #[`item.ruleVersion`]="{item}">
+          {{ shortStigId(item.ruleVersion) }}
+        </template>
+        <template #[`item.ruleId`]="{item}">
+          {{ shortRuleId(item.ruleId) }}
+        </template>
+        <template #[`item.cciRef`]="{item}">
+          {{ shortRuleId(item.cciRef) }}
+        </template>
+      </v-data-table>
+    </v-card-text>
+  </v-card>
+</template>
+
+<script lang="ts">
+import {ChecklistFilter, FilteredDataModule} from '@/store/data_filters';
+import {FileID} from '@/store/report_intake';
+import {ChecklistVuln} from '@mitre/hdf-converters';
+import {InspecDataModule} from '@/store/data_store';
+import _ from 'lodash';
+import {Component, Prop, Vue} from 'vue-property-decorator';
+
+@Component
+export default class ChecklistRulesTable extends Vue {
+  @Prop({type: Object, required: true}) readonly allFilter!: ChecklistFilter;
+  @Prop({type: Array, required: true}) readonly fileFilter!: FileID[];
+  @Prop({type: Boolean, required: true}) readonly shortIdEnabled!: boolean;
+  @Prop({type: Array, required: true}) readonly rules!: ChecklistVuln[];
+
+  shortRuleId(ruleId: string) {
+    if (this.shortIdEnabled) return ruleId.split('r')[0] || ruleId;
```

Comment:
Since we are using `vue-property-decorator`, moving the functions to a utility file might not clean up the file too much since they need to be declared in the `@Component` area to be used in the `template` tag.